pages/_app/index.jsx

```diff
@@ -21,12 +26,40 @@ class App extends NextApp {
     }
   }
 
-  store = null
+  redirectToLogin = () => {
+    window.location.replace('/login.html')
+  }
 
-  componentDidMount() {
+  reflectStoreToRoute() {
+    const { store } = this
     const { router } = this.props
-    const { dataProvider, activity } = new Route(router.asPath)
-    this.store = initStore({ dataProvider, activity })
+
+    if (store == null) return
+
+    const route = new Route({
+      dataProvider: store.dataProvider?.id,
+      activity: store.activity?.path,
+    })
+
+    // TODO: This probably should be under condition
+    router.push(route.href, route.as)
+  }
+
+  async componentDidMount() {
+    const store = initStore()
+
+    // Assumes store object is always the same
+    this.store = store
+    this.disposeRouter = autorun(this.reflectStoreToRoute.bind(this))
```

Comment:
I always considered using autorun outside of the store like a bad practice but I don't know why. :D

Reply:
In my opinion, `autorun` is a simple `observer`. Comparing to `computed` that runs on request, `autorun` runs always.

Maybe you have a point. I will think about it more.